src/core/agent/services/connectionService.ts

```diff
@@ -261,31 +260,57 @@ class ConnectionService extends AgentService {
     connectionId: string,
     note: ConnectionNoteProps
   ): Promise<void> {
-    await this.connectionNoteStorage.save({
-      id: uuidv4(),
-      title: note.title,
-      message: note.message,
-      connectionId,
+    const id = new Salter({}).qb64;
+    await this.props.signifyClient.contacts().update(connectionId, {
+      [`${ConnectionService.CONNECTION_NOTE_PREFIX_KEY}${id}`]: JSON.stringify({
+        ...note,
+        id: `${ConnectionService.CONNECTION_NOTE_PREFIX_KEY}${id}`,
+      }),
     });
   }
 
   async updateConnectionNoteById(
+    connectionId: string,
     connectionNoteId: string,
     note: ConnectionNoteProps
   ) {
-    const noteRecord = await this.connectionNoteStorage.findById(
-      connectionNoteId
-    );
-    if (!noteRecord) {
-      throw new Error(ConnectionService.CONNECTION_NOTE_RECORD_NOT_FOUND);
-    }
-    noteRecord.title = note.title;
-    noteRecord.message = note.message;
-    await this.connectionNoteStorage.update(noteRecord);
+    const connection = await this.props.signifyClient
+      .contacts()
+      .get(connectionId)
+      .catch((error) => {
+        const status = error.message.split(" - ")[1];
+        if (/404/gi.test(status)) {
+          throw new Error(`${Agent.MISSING_DATA_ON_KERIA}: ${connectionId}`);
+        } else {
+          throw error;
+        }
+      });
+    const connectionNote = JSON.parse(connection[connectionNoteId]);
+    connectionNote.title = note.title;
+    connectionNote.message = note.message;
+    await this.props.signifyClient.contacts().update(connectionId, {
+      ...connection,
+      [connectionNoteId]: JSON.stringify(connectionNote),
+    });
   }
 
-  async deleteConnectionNoteById(connectionNoteId: string) {
-    return this.connectionNoteStorage.deleteById(connectionNoteId);
+  async deleteConnectionNoteById(
+    connectionId: string,
+    connectionNoteId: string
+  ) {
+    const connection = await this.props.signifyClient
+      .contacts()
+      .get(connectionId)
+      .catch((error) => {
+        const status = error.message.split(" - ")[1];
+        if (/404/gi.test(status)) {
+          throw new Error(`${Agent.MISSING_DATA_ON_KERIA}: ${connectionId}`);
+        } else {
+          throw error;
+        }
+      });
+    delete connection[connectionNoteId];
+    return this.props.signifyClient.contacts().update(connectionId, connection);
```

Comment:
I haven't fully reviewed this because I was going to implement the delete functionality.

Does this actually work? Looking at keripy, `update` will still keep other fields already in the DB when you call `update`

Reply:
will write the PR for KERIA today

Reply:
ok easier to nullify